timelock-agent/src/main/java/com/palantir/timelock/config/TimeLockInstallConfiguration.java

```diff
@@ -18,12 +22,27 @@
 @JsonSerialize(as = ImmutableTimeLockInstallConfiguration.class)
 @Value.Immutable
 public interface TimeLockInstallConfiguration {
-    PaxosInstallConfiguration paxos();
+    Optional<PaxosInstallConfiguration> optionalPaxosConfig();
+
+    Optional<KeyValueServiceConfig> optionalKvsConfig();
 
     ClusterConfiguration cluster();
 
     @Value.Default
     default AsyncLockConfiguration asyncLock() {
         return ImmutableAsyncLockConfiguration.builder().build();
     }
+
+    @Value.Default
+    default PaxosInstallConfiguration paxos() {
+        return optionalPaxosConfig().orElseGet(() -> ImmutablePaxosInstallConfiguration.builder().build());
+    }
+
+    @Value.Check
+    default void check() {
+        Preconditions.checkArgument(optionalPaxosConfig().isPresent() || optionalKvsConfig().isPresent(),
```

Comment:
This check doesn't do what the message claims. I could have both a paxos and kvs config, and pass this check.
`optionalPaxosConfig().isPresent() != optionalKvsConfig().isPresent()`

Reply:
After thinking further, I think we still need the PaxosInstallConfig i.e the data directory. So, making it not optional.